gpflow/conditionals.py

```diff
@@ -129,31 +130,60 @@ def _conditional(Xnew, X, kern, f, *, full_cov=False, q_sqrt=None, white=False):
 # ----------------------------------------------------------------------------
 
 
-@sample_conditional.register(object, InducingFeature, Kernel, object)
+@sample_conditional.register(TensorArray, InducingFeature, Kernel, TensorArray)
 @name_scope("sample_conditional")
-def _sample_conditional(Xnew, feat, kern, f, *, full_output_cov=False, q_sqrt=None, white=False):
+def _sample_conditional(Xnew, feat, kern, f, *, full_cov=False, full_output_cov=False, q_sqrt=None, white=False, num_samples=None):
```

Comment:
Wouldn't it better to introduce `cov_structure` with different options, `full_cov, full_output_cov`, instead of introducing more and more parameters?

Reply:
that should be a separate PR :stuck_out_tongue: